pointcloud_boxcrop/CMakeLists.txt

```
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.8)
+project(pointcloud_boxcrop)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Wshadow -Wold-style-cast)
+endif()
+
+set(PACKAGE_DEPENDENCIES
+    ament_cmake
+    generate_parameter_library
+    pcl_conversions
+    pcl_ros
+    tf2
+    tf2_geometry_msgs
+    tf2_ros
+    tf2_sensor_msgs
+    rclcpp
+    sensor_msgs
+    vision_msgs)
+
+foreach(PACKAGE IN ITEMS ${PACKAGE_DEPENDENCIES})
+  find_package(${PACKAGE} REQUIRED)
+endforeach()
+
+find_package(PCL REQUIRED COMPONENTS filters)
+
+include_directories(${PCL_INCLUDE_DIRS})
+link_directories(${PCL_LIBRARY_DIRS})
+add_definitions(${PCL_DEFINITIONS})
+
+include_directories(include ${PCL_INCLUDE_DIRS})
```

Did you check if all this lines are necessary? Also you can merge `include_directories` into one.
```suggestion
include_directories(include ${PCL_INCLUDE_DIRS})
link_directories(${PCL_LIBRARY_DIRS})
add_definitions(${PCL_DEFINITIONS})
```